docs/flow/reference/components/prometheus.relabel.md

```diff
@@ -0,0 +1,181 @@
+of each metric in order of their appearance in the configuration file.
+
+Labels beginning with a double underscore are reserved for internal used and
+will be silently discarded after all relabeling steps have been applied.
+
+Multiple `prometheus.relabel` components can be specified by giving them
+different labels.
+
+## Example
+
+```river
+prometheus.relabel "keep_mysql_only" {
+  forward_to [prometheus.remote_write.onprem.receiver]
+
+  metric_relabel_config {
+    source_labels = ["__address__", "instance"]
+    separator     = "/"
+    target_label  = "host"
+    action        = "replace"
+  }
+
+  metric_relabel_config {
+    source_labels = ["app"]
+    action        = "keep"
+    regex         = "backend"
+  }
+
+  metric_relabel_config {
+    action = "labeldrop"
+    regex  = "instance"
+  }
+}
+```
+
+Let's see how the previous instance of the `prometheus.relabel` component would
+act on some metrics.
+
+```
+metric_a{__address__ = "localhost", instance = "development", app = "frontend"} 10
+metric_a{__address__ = "localhost", instance = "development", app = "backend"}	2
+metric_a{__address__ = "cluster_a", instance = "production",  app = "frontend"} 7
+metric_a{__address__ = "cluster_a", instance = "production",  app = "backend"}	9
+metric_a{__address__ = "cluster_b", instance = "production",  app = "database"}	4
+```
+
+After applying the first `metric_relabel_config` block, the `replace` action
+would populate a new label named `host` by the concatenating the two
+`__address__` and `instance` labels separated by a slash `/`.
+
+```
+metric_a{host = "localhost/development", __address__ = "localhost", instance = "development", app = "frontend"} 10
+metric_a{host = "localhost/development", __address__ = "localhost", instance = "development", app = "backend"}	2
+metric_a{host = "cluster_a/production",  __address__ = "cluster_a", instance = "production",  app = "frontend"} 7
+metric_a{host = "cluster_a/production",  __address__ = "cluster_a", instance = "production",  app = "backend"}	9
+metric_a{host = "cluster_b/production",  __address__ = "cluster_a", instance = "production",  app = "database"}	4
+```
+
+After applying the second relabeling step, the `keep` action would only keep
+the metrics whose `app` label match `regex`, dropping everything else, so the
+list of metrics would be trimmed down to 
+
+```
+metric_a{host = "localhost/development", __address__ = "localhost", instance = "development", app = "backend"}	2
+metric_a{host = "cluster_a/production",  __address__ = "cluster_a", instance = "production",  app = "backend"}	9
+```
+
+The third and final relabeling step which uses the `labeldrop` action would
+remove the `instance` label from the set of labels
+
+```
+metric_a{host = "localhost/development", __address__ = "localhost", app = "backend"}	2
+metric_a{host = "cluster_a/production",  __address__ = "cluster_a", app = "backend"}	9
+```
+
+Finally, the `__address__` label, as it starts with a double underscore and is
```

Comment:
This isn't quite right.

Reserved labels are removed from _targets_ prior to scraping, but never from metrics.